tests/test_utils.sh

```diff
@@ -694,6 +694,34 @@ function test_path_format_last_dir() {
   assertEquals "full directory formatting with multichar separator" "{l}pathname" "$lp_path_format"
 }
 
+function test_lp_fill {
```

Comment:
Should include a test with some Unicode characters, as well as some multi-byte Unicode characters and some extra wide ones.

I'm willing to bet that the extra wide ones will not work correctly. Not sure how to fix that, maybe it's not possible.

Reply:
On my terminal emulator, everything is "fine": wide characters are just overlapping each others and hiding the beginning of the next one.

If we allow for multiple filling characters, this can be mitigated, but a user would have to add extra space to the right part by herself and the spacing may be randomly inconsistent. I'll try that.

Reply:
Ah, I didn't think of Unicode characters as the fill string. I meant for the left or right strings, since we need to calculate their length. But the length we want is the printing size, not how many characters they have.

Reply:
How do we have access to this information? Shells only returns the number of characters when we ask for `${#string}`. I would find it easier to just let the user fill in the necessary blank spaces after a large unicode character (which is now possible, thanks to the multi-character filling support).

Reply:
That is the $1M question. This SO post asks the same: https://unix.stackexchange.com/q/245013/421569

The answer is that there is not a great way to do it. Our options are:
1. Use `wc -L`: `length=$(printf '%s' "${foo}" | wc -L)`. Being a subshell, this will be very slow. Also, not guaranteed to be accurate. 
2. Print the string to the terminal, and record the position before and after. This is the most likely to be accurate. Likely would have a few issues:
    * Reading from the terminal emulator can be buggy, and might hang and lock up.
    * Printing text to the terminal is not ideal. If we reset the cursor position after, when PS1 is printed it should write over these characters, meaning we don't need to worry about it.
    * If *anything* else printed to the terminal at the same time we did, it could give a terribly wrong answer.
    * Portability issues?
3. Use an `awk` script. Doesn't require a subshell, but does require a binary call.
4. Just don't. Multi-width or zero width characters are somewhat rare, so take the easy route and ignore the problem.

I'm not sure what the right answer is, but probably we should try a few of them and see how well they work.

Note that this is something I've been thinking about for a while, as it's needed to solve #599 as well.

Reply:
I would go for 4, given that it's almost just a templating issue, at the end. If you configure your prompt to use long chars, you will see the results pretty easily. And on all the terminal emulator I have on my box, long characters are just overwritten on top of each others, so that I just have to add spaces, if I want to match their width. The only issue remaining would actually be the very end of line, indeed, but that's just *one* (long) character, and probably a very rare case.